spec/integration/posts_index_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe 'posts#index', type: :feature do
+  describe 'Post' do
+    before(:each) do
+      @user1 = User.create(name: 'Ranj', photo: 'Ranj.jpeg', bio: 'bio', posts_counter: 0)
+      @user1.save!
+      visit root_path
+
+      @post1 = Post.create(title: 'First Post', text: 'This is my first post', comments_counter: 0, likes_counter: 3,
+                           author: @user1)
+      @post2 = Post.create(title: 'Second Post', text: 'This is my second post', comments_counter: 0, likes_counter: 0,
+                           author: @user1)
+      @post3 = Post.create(title: 'Third Post', text: 'This is my third post', comments_counter: 0, likes_counter: 0,
+                           author: @user1)
+      @comment1 = Comment.create(text: 'Comment first', author: User.first,
+                                 post: @post1)
+      @comment2 = Comment.create(text: 'Comment another', author: User.first, post: @post1)
+      @comment3 = Comment.create(text: 'Comment new', author: User.first, post: @post1)
+    end
+
+    it "shows user's profile picture" do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_css('img[src*="Ranj.jpeg"]')
+    end
+
+    it 'shows the users username' do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_content('Ranj')
+    end
+
+    it 'shows number of posts of user has written' do
+      visit(user_posts_path(@user1.id))
+      expect(@user1.posts_counter).to eql(3)
+    end
+
```

Comment:
- [ ] The post index page is missing the below test as requested in the project requirements. Please make sure to add that and should pass without any errors.
  - Tests that asserts you can see a section for pagination if there are more posts than fit on the view.